game-core/src/main/java/games/strategy/triplea/odds/calculator/BattleCalculatorPanel.java

```diff
@@ -1413,6 +1418,7 @@ private void setWidgetActivation() {
       final Collection<TerritoryEffect> territoryEffects = getTerritoryEffects();
       final IntegerMap<UnitType> costs = TuvUtils.getCostsForTuv(getAttacker(), data);
       attackers.sort(new UnitBattleComparator(false, costs, territoryEffects, data).reversed());
+      final Territory location = findPotentialBattleSite();
```

Comment:
Any chance location will already be computed? Is the instance `location` attached to the class still used at all, or would it be better to get rid of that state and do more on-demand computing?

Reply:
Yes, location is already computed if you click on a territory and use the shortcut to open the battle calculator.  The way to get a null location is by opening it from the menu.

In the case where you are opening the battle calculator from a specific territory, we want to use that territory.  I believe the calculator utilizes territory attachments for its calculations so you'll get more accurate results.